horovod/spark/keras/remote.py

```diff
@@ -236,56 +225,42 @@ def train(serialized_model, train_rows, val_rows, avg_row_size):
-                                workers_count=train_reader_worker_count,
-                                shard_count=hvd.size(),
-                                hdfs_driver=PETASTORM_HDFS_DRIVER,
-                                schema_fields=schema_fields,
-                                transform_spec=transform_spec,
-                                storage_options=storage_options,
-                                shuffle_rows=shuffle,
-                                shuffle_row_groups=shuffle,
-                                seed=random_seed,
-                                **reader_factory_kwargs) as train_reader:
-                with reader_factory(remote_store.val_data_path,
-                                    num_epochs=1,
-                                    cur_shard=hvd.rank(),
-                                    reader_pool_type=reader_pool_type,
-                                    workers_count=val_reader_worker_count,
-                                    shard_count=hvd.size(),
-                                    hdfs_driver=PETASTORM_HDFS_DRIVER,
-                                    schema_fields=schema_fields,
-                                    transform_spec=transform_spec,
-                                    storage_options=storage_options,
-                                    shuffle_rows=False,
-                                    shuffle_row_groups=False,
-                                    **reader_factory_kwargs) \
-                    if should_validate else empty_batch_reader() as val_reader:
-
-                    train_data = make_dataset(train_reader, batch_size,
-                                              is_batch_reader, shuffle=shuffle, cache=inmemory_cache_all)
-                    val_data = make_dataset(val_reader, val_batch_size,
-                                            is_batch_reader, shuffle=False, cache=inmemory_cache_all) \
-                        if val_reader else None
-
-                    history = fit(model, train_data, val_data, steps_per_epoch,
-                                  validation_steps, callbacks, verbose)
+
+            data_module_kwargs = {
+                'label_cols': label_columns,                    # nvtabular
+                'continuous_cols': continuous_columns,          # nvtabular
+                'categorical_cols': categorical_columns,        # nvtabular
+                'make_dataset': make_dataset,                   # petastorm
+                'random_seed': random_seed,                     # petastorm
+                'train_dir': remote_store.train_data_path,
+                'val_dir': remote_store.val_data_path,
+                'num_train_epochs': epochs,
+                'has_val': should_validate is not None,
+                'train_batch_size': batch_size,
+                'val_batch_size': val_batch_size,
+                'shuffle': shuffle,
+                'random_seed': random_seed,
+                'num_reader_epochs': epochs,
```

Comment:
`num_reader_epochs` is not used in Petastorm for Keras/TF, remove it.
You see that we use` epoch=1` and `loop infinitely` in dataloader to avoid introducing duplicated samples in every epoch.
I think you got it from Lightning datamodule usage. Sorry about confusion.

I agree Petastorm has difference of Keras/TF and Lightning, I will try to clean them a little bit after merging this PR.

Reply:
Also removed `debug_data_loader`, `train_async_data_loader_queue_size`, and `val_async_data_loader_queue_size`, since these were similar.

Reply:
sure.